test/e2e/framework.go

```diff
@@ -521,23 +594,25 @@ func (data *TestData) runPingCommandFromTestPod(namespace string, podName string
 	return nil
 }
 
-func (data *TestData) runNetcatCommandFromTestPod(namespace string, podName string, server string, port int) error {
-	// Retrying several times to avoid flakes as the test may involve DNS (coredns) and Service/Endpoints (kube-proxy).
+func (data *TestData) runNetcatCommandFromPod(namespace string, podName string, containerName string, server string, port int) error {
 	cmd := []string{
 		"/bin/sh",
 		"-c",
-		fmt.Sprintf("for i in $(seq 1 5); do nc -vz -w 4 %s %d && exit 0 || sleep 1; done; exit 1",
+		fmt.Sprintf("for i in $(seq 1 5); do nc -w 4 %s %d && exit 0 || sleep 1; done; exit 1",
 			server, port),
 	}
-	stdout, stderr, err := data.runCommandFromPod(namespace, podName, podName, cmd)
-	if err == nil {
-		return nil
+	_, _, err := data.runCommandFromPod(namespace, podName, containerName, cmd)
+	if err != nil {
+		return err
 	}
-	return fmt.Errorf("nc stdout: <%v>, stderr: <%v>, err: <%v>", stdout, stderr, err)
+	return nil
 }
 
 func applyYAML(filename string, ns string) error {
 	cmd := fmt.Sprintf("kubectl apply -f %s -n %s", filename, ns)
+	if ns == "" {
+		cmd = fmt.Sprintf("kubectl apply -f %s", filename)
+	}
 	var stdout, stderr bytes.Buffer
 	command := exec.Command("bash", "-c", cmd)
 	log.Printf("Applying YAML file %s", cmd)
```

Comment:
it should be filename here instead of cmd?

Reply:
good